sdl_android/src/main/java/com/smartdevicelink/api/VideoStreamingManager.java

```diff
@@ -0,0 +1,571 @@
+						VideoStreamingParameters params = new VideoStreamingParameters();
+						params.update((VideoStreamingCapability)capability);	//Streaming parameters are ready time to stream
+						startVideoStreaming(params, encrypted);
+					}
+
+					@Override
+					public void onError(String info) {
+						stateMachine.transitionToState(StreamingStateMachine.ERROR);
+						Log.e(TAG, "Error retrieving video streaming capability: " + info);
+					}
+				});
+			}else{
+				//We just use default video streaming params
+				VideoStreamingParameters params = new VideoStreamingParameters();
+				DisplayCapabilities dispCap = (DisplayCapabilities)internalInterface.getCapability(SystemCapabilityType.DISPLAY);
+				if(dispCap !=null){
+					params.setResolution(dispCap.getScreenParams().getImageResolution());
+				}
+				startVideoStreaming(params, encrypted);
+			}
+		}else{
+			startVideoStreaming(parameters, encrypted);
+		}
+	}
+
+	/**
+	 * Opens the video service (serviceType 11) and creates a Surface (used for streaming video) with input parameters provided by the app
+	 * @param frameRate - specified rate of frames to utilize for creation of Surface
+	 * @param iFrameInterval - specified interval to utilize for creation of Surface
+	 * @param width - specified width to utilize for creation of Surface
+	 * @param height - specified height to utilize for creation of Surface
+	 * @param bitrate - specified bitrate to utilize for creation of Surface
+	 *@return Surface if service is opened successfully and stream is started, return null otherwise
+	 */
+	@SuppressWarnings("unused")
+	public Surface createOpenGLInputSurface(int frameRate, int iFrameInterval, int width,
+	                                        int height, int bitrate, boolean isEncrypted) {
+		if(hmiLevel != HMILevel.HMI_FULL){
+			Log.e(TAG, "Cannot start video service if HMILevel is not FULL.");
+			return null;
+		}
+
+		IVideoStreamListener encoderListener = startVideoStream(new VideoStreamingParameters(), isEncrypted);
+		if (encoderListener == null) {
+			stateMachine.transitionToState(StreamingStateMachine.ERROR);
+			Log.e(TAG, "Cannot create encoderListener");
+			return null;
+		}
+
+		mSdlEncoder = new SdlEncoder();
+		mSdlEncoder.setFrameRate(frameRate);
+		mSdlEncoder.setFrameInterval(iFrameInterval);
+		mSdlEncoder.setFrameWidth(width);
+		mSdlEncoder.setFrameHeight(height);
+		mSdlEncoder.setBitrate(bitrate);
+		mSdlEncoder.setOutputListener(encoderListener);
+		Surface surface = mSdlEncoder.prepareEncoder();
+
+		if(surface != null){
+			stateMachine.transitionToState(StreamingStateMachine.READY);
+		}else{
+			Log.e(TAG, "Cannot create surface.");
+			stateMachine.transitionToState(StreamingStateMachine.ERROR);
+		}
+		return surface;
+	}
+
+	/**
+	 * Starts the MediaCodec encoder utilized in conjunction with the Surface returned via the createOpenGLInputSurface method
+	 * @see #createOpenGLInputSurface(int, int, int, int, int, boolean)
+	 */
+	public void startEncoder(){
```

Comment:
It seems that these are exposed to the developer and they are responsible for handling them. I would assume the manager would take care of this instead. I'm wondering if allowing this is even useful for developers since they could create an `SdlRemoteDisplay` that contains a surface view on which they could draw.

Reply:
Should I just call this internally in `createOpenGLInputSurface` after creating the surface then?

Reply:
As discussed offline, if we remove the `createOpenGLInputSurface` functionality for this release these methods can also be removed. We will instruct developers to create a remote display with a surface view for now.